test/test_geocaching.py

```diff
@@ -96,6 +96,57 @@ def test_search_quick_match_load(self):
                 except PMOnlyException:
                     pass
 
+    def test_search_rect(self):
+        """Perform search by rect and check found caches"""
+        rect = Rectangle(Point(49.73, 13.38), Point(49.74, 13.39))
+
+        with self.subTest("default use"):
+            with self.recorder.use_cassette('geocaching_search_rect'):
+                gc = Geocaching(session=self.session)
+                gc.login(_username, _password)
+                caches = self.gc.search_rect(rect)
+                expected = {cache.wp for cache in caches}
+
+                self.assertEqual(expected, {
+                    'GC1TYYG', 'GC11PRW', 'GC7JRR5', 'GC161KR', 'GC1GW54', 'GC7KDWE', 'GC8D303'
+                })
+
+        with self.subTest("sort by distance"):
+            with self.recorder.use_cassette('geocaching_search_rect_by_distance'):
+                with self.assertRaises(AssertionError):
+                    caches = list(self.gc.search_rect(rect, sortby='distance'))
+
+                origin = Point.from_string('N 49° 44.230 E 013° 22.858')
+
+                caches = list(self.gc.search_rect(rect, sortby='distance', origin=origin))
+
+                expected = [cache.wp for cache in caches]
+                self.assertEqual(expected, [
+                    'GC11PRW', 'GC1TYYG', 'GC7JRR5', 'GC1GW54', 'GC161KR', 'GC7KDWE', 'GC8D303'
+                ])
+
+                # Check if caches are sorted by distance to origin
+                distances = [great_circle(cache.location, origin).meters for cache in caches]
+                self.assertEqual(distances, sorted(distances))
+
+    def test_api_rate_limit(self):
+        """Test recovering from API rate limit exception."""
+        # WARNING: casset for this test has been manualy edited!
```

Comment:
There is a typo in this comment.

And could you please elaborate on the changes here to allow re-recording and modifying the cassette if required?

Reply:
I have simplified this testcase by removing any manual editing, cassette is quite big, but should be fully reproducible.

Reply:
With the change it is stated that one has to be perform a lot of requests. I am not sure about this and how to correctly handle it.

Reply:
@tomasbedrich What do you think about this specific test case?

Reply:
I understand that re-recording this cassette is expensive, but I expect that we do not have to do it more often than a few times a year. Therefore, I see this solution as good enough. If it turns to be a problem, we can always switch to mocking. Is there anything else which I miss?

Reply:
I discovered that rate limits refer to number of performed request, but not to number of downloaded caches, so I limited number of caches downloaded in each requests and cassettes are small now :) I also increased default `per_query` limit to `200`.